src/main/java/com/iexec/core/replicate/ReplicateSupplyService.java

```diff
@@ -173,43 +173,51 @@ private boolean isFewBlocksAfterInitialization(Task task) {
             String enclaveChallenge = smsService.getEnclaveChallenge(chainTaskId, task.isTeeNeeded());
             if (task.isTeeNeeded() && enclaveChallenge.isEmpty()) continue;
 
-            Optional<RecoveryAction> oRecoveryAction = getAppropriateRecoveryAction(task, replicate, blockNumber);
-            if (!oRecoveryAction.isPresent()) continue;
+            Optional<TaskNotificationType> optionalTaskNotificationType = getAppropriateRecoveryAction(task, replicate, blockNumber);
+            if (!optionalTaskNotificationType.isPresent()) continue;
 
             // generate contribution authorization
             ContributionAuthorization authorization = signatureService.createAuthorization(
                     walletAddress, chainTaskId, enclaveChallenge);
 
-            InterruptedReplicateModel interruptedReplicate = InterruptedReplicateModel.builder()
+            //TODO Send only required extras
+            TaskNotificationExtra taskNotificationExtra = TaskNotificationExtra.builder()
                     .contributionAuthorization(authorization)
-                    .recoveryAction(oRecoveryAction.get())
+                    .blockNumber(task.getConsensusReachedBlockNumber())
```

Comment:
We shouldn't send the blockNumber when the notif is pleas_contribute (consensus is not yet reached)

Reply:
Done